docs/tutorial/creating_cli.md

```diff
@@ -0,0 +1,622 @@
+
+```scala 3
+def addCommand: IO[Unit] =
+  for 
+    username    <- IO.println("Enter the username: ") >> IO.readLine
+    firstName   <- IO.println("Enter the first name: ") >> IO.readLine
+    lastName    <- IO.println("Enter the last name: ") >> IO.readLine
+    phoneNumber <- IO.println("Enter the phone number: ") >> IO.readLine
+    email       <- IO.println("Enter the email: ") >> IO.readLine
+
+    contact = Contact(username, firstName, lastName, phoneNumber, email)
+    
+    _ <- cm.addContact(contact)
+      .flatMap(username => IO.println(s"Contact $username added"))
+      .handleErrorWith:
+        case ContactFound(username) =>
+          IO.println(s"Contact $username already exists")
+        case e =>
+          IO.println(s"An error occurred: \n${e.printStackTrace()}")
+    
+  yield ()
+```
+
+The first five lines after the `for` asks the fields to save the contact. Then we create the `Contact` object with the information provided. Next, we call the `addContact` method of the `ContactManager` and handle the errors. If the contact already exists, we print a message to the user. If another error occurs, we print the stack trace of the error just to see what went wrong.
+
+Now the `removeCommand` function:
+
+```scala 3 
+def removeCommand(username: Username): IO[Unit] =
+  cm.removeContact(username) >> IO.println(s"Contact $username removed")
+```
+It simply calls the `removeContact` function and print a message to the user. The function doesn't corroborate if the contact exists and will delete any matching username anyway, but you can add this logic if you want.
+
+The `searchIdCommand` function is also straight forward, matches the `Options` returned by the `searchId` method and prints the user information if it exists:
+
+```scala 3
+def searchIdCommand(username: Username): IO[Unit] =
+  for
+    contact <- cm.searchId(username)
+    _ <- contact match 
+      case Some(c) => IO.println(c.show)
+      case None    => IO.println(s"Contact $username not found")
+  yield ()
+```
+
+Like the previous `searchX` variants, they’re fairly similar, so we will only show one of them.
+In this case, the `searchEmailCommand` one:
+
+```scala 3
+def searchEmailCommand(email: Email): IO[Unit] =
+  for {
+    contacts <- cm.searchEmail(email)
+    _        <- contacts.traverse_(c => IO.println(c.show))
+} yield ()
+``` 
+
+It also uses the `traverse_` method, but with a little variant that optimises it a bit. It takes as a parameter a function from `A => IO[Unit]` and it will execute the effect for each element of the collection, but it will discard the result of the computation.
+
+Last but not least, the `updateCommand` function. The idea here is to take as parameters the username to change the contact and a list of flags parsed from the command line that contain information about which field to update and the new value of the field, and apply each of them:
+
```

Comment:
```suggestion
Last but not least, the `updateCommand` function. The idea here is to take as parameters the username to change the contact and a list of flags parsed from the command line that contain information about which field to update and the new value of the field, and apply each of them:

```
This is a pretty long and hard to parse line, can you break the second sentence down a bit more, what are the parameters and to what does it apply each of them?